### PR TITLE
Change PageController#nextPage & #previousPage methods to return Future when the animation completes.

### LINKED ISSUE #12298: PageController #nextPage & #previousPage should return a Future, but the method signature is currently void.
Specific lines for the [#nextPage method](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/page_view.dart#L116-L120) and the [#previousPage method](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/page_view.dart#L126-L130).

The Dart docs advertise that these methods return a `Future`, but the signature is `void` instead.

Also, [the #jumpToPage method advertises](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/page_view.dart#L104-L107) that it returns a `Future`, but it has a `void` return type. Seems this is working the intended way, but the docs are wrong.

### PR DESCRIPTION
**Pull request by roughike**
Fixes #12298.

If there should be tests to go along with it, please let me know. Wasn't sure of the place in `page_view_test.dart` where to test this.

### CODE REVIEW

**Comment by roughike** [comment]
Oopsie daisie, did this at work (I'm allowed to), but haven't signed the CLA on my work email account. Will resolve this in one hour or so and notify about it.

**Comment by Hixie** [comment]
This seems reasonable but we should add some tests.

**Comment by Hixie** [comment]
(The tests are to make sure that we don't accidentally revert this change. Basically, you want to make sure that everything you care about is tested so that if we randomly start changing code, we won't break what you care about.)

**Comment by roughike** [comment]
I tried to test this the most logical way, which would be something like:

```dart
await controller.nextPage(duration: const Duration(milliseconds: 150), curve: Curves.ease);
expect(controller.page, 3);
```

However, this produced a `TimeoutException` (possibly related to #5728), so I just resorted to checking the method signature. Let me know if there's a better way.

(Give me a min and I'll fix the merge conflict as well.)

**Comment by Hixie** [comment]
Ah yes, you can't `await` a Future from the framework in tests (because time stands still for the framework except when you call pump, so the future never resolves!). You have to take the Future, call `.then` on it, then call `tester.pump()` with an appropriate duration a few times, checking that the `.then()` callback only runs when you expect it (e.g. set a flag in the callback, verify the flag is false before and after the pumps except for the last one, and true after the last pump that takes the time across the line).

**Comment by roughike** [comment]
Thanks for pointing me to the right direction!

See if the tests makes more sense now.

File: packages/flutter/test/widgets/page_view_test.dart [hunk 1]
```diff
@@ -209,6 +209,38 @@ void main() {
...
+    controller.nextPage(duration: const Duration(milliseconds: 150), curve: Curves.ease)
+        .then((_) => nextPageCompleted = true);
+
+    expect(nextPageCompleted, false);
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(nextPageCompleted, false);
+    await tester.pump(const Duration(milliseconds: 200));
+    expect(nextPageCompleted, true);
+
+
```
Comment (abarth) [inline packages/flutter/test/widgets/page_view_test.dart]: extra blank line here

**Comment by abarth** [comment]
LGTM

**Pull request merged**